### PR TITLE
Enhance Test Coverage for @yamada-ui/select

### LINKED ISSUE #2072: Enhance Test Coverage for `@yamada-ui/select`
package: @yamada-ui/select

### Description
`@yamada-ui/select` has a test coverage of less than 95%. We will enhance it to above 95%.

### Target files
- [packages/components/select/src/multi-select.tsx](https://app.codecov.io/gh/yamada-ui/yamada-ui/blob/main/packages/components/select/src/multi-select.tsx)
  - [L261](https://app.codecov.io/gh/yamada-ui/yamada-ui/blob/main/packages/components/select/src/multi-select.tsx#L261)
- [packages/components/select/src/use-select.ts](https://app.codecov.io/gh/yamada-ui/yamada-ui/blob/main/packages/components/select/src/use-select.ts)
  - [L261](https://app.codecov.io/gh/yamada-ui/yamada-ui/blob/main/packages/components/select/src/use-select.ts#L261)
  - [L282](https://app.codecov.io/gh/yamada-ui/yamada-ui/blob/main/packages/components/select/src/use-select.ts#L282)
  - [L314](https://app.codecov.io/gh/yamada-ui/yamada-ui/blob/main/packages/components/select/src/use-select.ts#L314)
  - [L370](https://app.codecov.io/gh/yamada-ui/yamada-ui/blob/main/packages/components/select/src/use-select.ts#L370)
  - [L396](https://app.codecov.io/gh/yamada-ui/yamada-ui/blob/main/packages/components/select/src/use-select.ts#L396)
  - [L408](https://app.codecov.io/gh/yamada-ui/yamada-ui/blob/main/packages/components/select/src/use-select.ts#L408)
  - [L463](https://app.codecov.io/gh/yamada-ui/yamada-ui/blob/main/packages/components/select/src/use-select.ts#L463)
  - [L482](https://app.codecov.io/gh/yamada-ui/yamada-ui/blob/main/packages/components/select/src/use-select.ts#L482)
  - [L500](https://app.codecov.io/gh/yamada-ui/yamada-ui/blob/main/packages/components/select/src/use-select.ts#L500)
  - [L505](https://app.codecov.io/gh/yamada-ui/yamada-ui/blob/main/packages/components/select/src/use-select.ts#L505)
  - [L510](https://app.codecov.io/gh/yamada-ui/yamada-ui/blob/main/packages/components/select/src/use-select.ts#L510)
  - [L515](https://app.codecov.io/gh/yamada-ui/yamada-ui/blob/main/packages/components/select/src/use-select.ts#L515)
  - [L562](https://app.codecov.io/gh/yamada-ui/yamada-ui/blob/main/packages/components/select/src/use-select.ts#L562)
  - [L715](https://app.codecov.io/gh/yamada-ui/yamada-ui/blob/main/packages/components/select/src/use-select.ts#L715)
  - [L770](https://app.codecov.io/gh/yamada-ui/yamada-ui/blob/main/packages/components/select/src/use-select.ts#L770)
  - [L925](https://app.codecov.io/gh/yamada-ui/yamada-ui/blob/main/packages/components/select/src/use-select.ts#L925)
  - [L931](https://app.codecov.io/gh/yamada-ui/yamada-ui/blob/main/packages/components/select/src/use-select.ts#L931)

### Problem Statement/Justification
Testing is essential to ensure quality when operating a package.

### Additional Information
While it's ideal to cover all tests, if there are too many lines to work on, please create a PR within your capacity.
Any deficiencies not covered by the PR will be automatically created as issues.

### PR DESCRIPTION
**Pull request by hinataOhyama**
Closes #2072

## Description

This pull request enhances the test coverage for the @yamada-ui/select package, specifically for the MultiSelect component. It adds a new test case to verify the behavior of the MultiSelect component when no props are passed and checks the focus handling.

## New behavior

A new test case has been added to multi-select.test.tsx that:

Renders the MultiSelect component with no props
Verifies that the component renders correctly with an empty span
Checks that the component is not focused initially
Ensures that clicking the component does not change its focus state
Confirms that no options are displayed when the component is empty
Validates that the component can receive and lose focus programmatically

### CODE REVIEW

File: packages/components/select/tests/multi-select.test.tsx [hunk 1]
```diff
@@ -0,0 +1,398 @@
...
+      await user.click(option1)
+      await user.click(option2)
+
+      await user.click(document.body)
+      await user.click(input)
+
+      expect(option1).toHaveAttribute("aria-checked", "true")
+      expect(option2).toHaveAttribute("aria-checked", "true")
+    })
+  })
```
Comment (108yen) [inline packages/components/select/tests/multi-select.test.tsx:165]: I did not see much need for this test.

**Review by 108yen — CHANGES_REQUESTED** [review]

File: packages/components/select/tests/multi-select.test.tsx (hunk 1 above)
Comment (hinataOhyama) [reply packages/components/select/tests/multi-select.test.tsx:165]: Thanks for the review!
Are you saying that I don't need to implement this test for line 261?
If so, what approach would you 108yen take to increase test coverage for use-select.ts?

Comment (108yen) [reply packages/components/select/tests/multi-select.test.tsx:165]: These features have not been tested.
I am sure there are many others.

- [set-maximum-number-of-selectable-values](https://yamada-ui.com/components/forms/multi-select#set-maximum-number-of-selectable-values)
- [close-dropdown-list-on-selection](https://yamada-ui.com/components/forms/multi-select#close-dropdown-list-on-selection)

Comment (hinataOhyama) [reply packages/components/select/tests/multi-select.test.tsx:165]: Understood.
Surely I should have created a test that tests the behavior of each feature,
not one that tests for coverage!

**Comment by hinataOhyama** [comment]
Is the work plan correct? Please re-review!

File: packages/components/select/tests/multi-select.test.tsx [hunk 2]
```diff
@@ -1,6 +1,23 @@
...
+  )
+
+  const input = screen.getByRole("combobox")
+  await user.click(input)
+
+  return { user, input }
+}
+
+const getOptions = () => screen.findAllByRole("option")
+
```
Comment (108yen) [inline packages/components/select/tests/multi-select.test.tsx:20]: It is difficult to understand, so please do the same as any other test.
Render the component in `test`.

File: packages/components/select/tests/multi-select.test.tsx [hunk 3]
```diff
@@ -134,33 +151,55 @@ describe("<MultiSelect />", () => {
...
       )
+      const { user } = await setupMultiSelect({
+        component: CustomComponent,
+      })
 
-      const input = screen.getByRole("combobox")
-      expect(input).toBeInTheDocument()
-      await user.click(input)
+      const options = await getOptions()
+      const option1 = options[0]
```
Comment (108yen) [inline packages/components/select/tests/multi-select.test.tsx:171]:
```tsx
const option1 = getByRole("option", {name: /one/i})
```

File: packages/components/select/tests/multi-select.test.tsx [hunk 4]
```diff
@@ -134,33 +151,55 @@ describe("<MultiSelect />", () => {
...
-      expect(option2).toBeVisible()
+      await user.click(customOption)
+      expect(customOption).toBeValid()
+    })
+
+    test("closeOnSelect should work correctly", async () => {
+      const { user, input } = await setupMultiSelect({ closeOnSelect: true })
 
+      const options = await getOptions()
+      const option1 = options[0]
```
Comment (108yen) [inline packages/components/select/tests/multi-select.test.tsx:185]:
```tsx
const option1 = getByRole("option", {name: /one/i})
```

**Review by 108yen — CHANGES_REQUESTED** [review]

File: packages/components/select/tests/multi-select.test.tsx (hunk 2 above)
Comment (hinataOhyama) [reply packages/components/select/tests/multi-select.test.tsx]: I understand!
I'll fix it!

**Review by 108yen — APPROVED** [review]

**Review by hirotomoyamada — APPROVED** [review]

**Pull request merged**